From f8e175fd0f6ba81471be9836279da1090b49e066 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 24 Nov 2022 14:25:56 +0200
Subject: [PATCH] [FW-0] gaudi3: fix debug straps numbers

Debug straps numbers were set according to the gpio excel numbering,
which is wrong. numbers are aligned to SocOnline strap register
fields.

Change-Id: I595f274b935a58884d3a909aea0edcd88e654360
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index cb53567..bdd9fae 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -277,13 +277,13 @@
 
 	switch (func) {
 	case HL_DEBUG_HW_PCIE:
-		strap_mask = BIT(5); /* BTL_DISABLE */
+		strap_mask = BIT(28); /* BTL_DISABLE */
 		break;
 	case HL_DEBUG_HW_DRAM:
-		strap_mask = BIT(4); /* RSVD[1] */
+		strap_mask = BIT(31); /* RSVD[1] */
 		break;
 	case HL_DEBUG_HW_I2C:
-		strap_mask = BIT(2); /* RSVD[0] */
+		strap_mask = BIT(30); /* RSVD[0] */
 		break;
 	default:
 		preboot_puts_hex("Unknown H/W function ", func, false);
